src/coreclr/src/System.Private.CoreLib/src/Internal/Runtime/InteropServices/ComponentActivator.cs

```diff
@@ -81,15 +86,7 @@ private static string MarshalToString(IntPtr arg, string argName)
                     throw new ArgumentNullException(nameof(functionHandle));
                 }
 
-                Delegate d = CreateDelegate(assemblyPath, typeName, methodName, delegateType);
-
-                IntPtr functionPtr = Marshal.GetFunctionPointerForDelegate(d);
-
-                lock (s_delegates)
-                {
-                    // Keep a reference to the delegate to prevent it from being garbage collected
-                    s_delegates[functionPtr] = d;
-                }
+                IntPtr functionPtr = GetFunctionPointer(assemblyPath, typeName, methodName, delegateType);
 
                 Marshal.WriteIntPtr(functionHandle, functionPtr);
```

Comment:
Nit: `Marsal.WritePtr` is just an expensive wrapper around `*(IntPtr*)p = value`. It is better to write the assignment directly.